tortoise/util.go

```diff
@@ -1,79 +1,95 @@
 package tortoise
 
 import (
+	"errors"
 	"fmt"
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
-type vec [2]int
+type vec [2]uint64
 
-const (
-	globalThreshold = 0.6
-)
-
-var ( //correction vectors type
-	// Opinion vectors
-	support = vec{1, 0}
-	against = vec{0, 1}
-	abstain = vec{0, 0}
+var (
+	support     = vec{1, 0}
+	against     = vec{0, 1}
+	abstain     = vec{0, 0}
+	errOverflow = errors.New("vector arithmetic overflow")
 )
 
 // Field returns a log field. Implements the LoggableField interface.
 func (a vec) Field() log.Field {
-	return log.String("vote_vector", fmt.Sprint(a))
+	return log.String("vote_vector", fmt.Sprintf("%s (+%d, -%d)", a, a[0], a[1]))
 }
 
 func (a vec) Add(v vec) vec {
-	return vec{a[0] + v[0], a[1] + v[1]}
+	a[0] += v[0]
+	a[1] += v[1]
+	// prevent overflow/wraparound
+	if a[0] < v[0] || a[1] < v[1] {
```

Comment:
maybe we could check before adding instead of check that overflow has happened?

Reply:
not sure what you mean by "check before adding", what's the difference? you mean like turn this into an error in the code that calls Add?

Reply:
I was thinking preventing the overflow before it happened but I know see there'
s not much difference. resolving this.